languages/rlang/src/main/java/de/jplag/rlang/RListener.java

```diff
@@ -0,0 +1,105 @@
+import static de.jplag.rlang.RTokenType.WHILE_BEGIN;
+import static de.jplag.rlang.RTokenType.WHILE_END;
+
+import de.jplag.antlr.AbstractAntlrListener;
+import de.jplag.rlang.grammar.RParser.ArrayAccessContext;
+import de.jplag.rlang.grammar.RParser.AssignmentContext;
+import de.jplag.rlang.grammar.RParser.BreakContext;
+import de.jplag.rlang.grammar.RParser.ComplexContext;
+import de.jplag.rlang.grammar.RParser.CompoundStatementContext;
+import de.jplag.rlang.grammar.RParser.FalseContext;
+import de.jplag.rlang.grammar.RParser.FloatContext;
+import de.jplag.rlang.grammar.RParser.ForContext;
+import de.jplag.rlang.grammar.RParser.FunctionCallContext;
+import de.jplag.rlang.grammar.RParser.FunctionDefinitionContext;
+import de.jplag.rlang.grammar.RParser.HelpContext;
+import de.jplag.rlang.grammar.RParser.HexContext;
+import de.jplag.rlang.grammar.RParser.IfContext;
+import de.jplag.rlang.grammar.RParser.IfElseContext;
+import de.jplag.rlang.grammar.RParser.IntContext;
+import de.jplag.rlang.grammar.RParser.ListAccessContext;
+import de.jplag.rlang.grammar.RParser.NamespaceAccessContext;
+import de.jplag.rlang.grammar.RParser.NextContext;
+import de.jplag.rlang.grammar.RParser.RepeatContext;
+import de.jplag.rlang.grammar.RParser.StringContext;
+import de.jplag.rlang.grammar.RParser.SubContext;
+import de.jplag.rlang.grammar.RParser.TrueContext;
+import de.jplag.rlang.grammar.RParser.WhileContext;
+
+/**
+ * Contains mapping for RLang to create JPlag tokens from ANTLR tokens
+ */
+public class RListener extends AbstractAntlrListener {
+    public RListener() {
+        header();
+        functions();
+        literals();
+        controlStructures();
+        assignments();
+    }
+
+    private void header() {
+        visit(ArrayAccessContext.class).mapRange(INDEX);
+        visit(ListAccessContext.class).mapRange(INDEX);
+        visit(NamespaceAccessContext.class).map(PACKAGE);
+    }
+
+    private void functions() {
+        visit(FunctionDefinitionContext.class).map(BEGIN_FUNCTION, END_FUNCTION);
+        visit(FunctionCallContext.class).mapRange(FUNCTION_CALL);
+
+        visit(CompoundStatementContext.class).map(COMPOUND_BEGIN, COMPOUND_END);
+        visit(HelpContext.class).map(HELP);
+    }
+
+    private void literals() {
+        visit(HexContext.class).map(NUMBER);
+        visit(IntContext.class).map(NUMBER);
+        visit(FloatContext.class).map(NUMBER);
+        visit(ComplexContext.class).map(NUMBER);
+        visit(StringContext.class).map(STRING);
+        visit(TrueContext.class).map(BOOL);
+        visit(FalseContext.class).map(BOOL);
+    }
+
+    private void controlStructures() {
+        visit(IfContext.class).map(IF_BEGIN, IF_END);
+        visit(IfElseContext.class).map(IF_BEGIN, IF_END);
+        visit(ForContext.class).map(FOR_BEGIN, FOR_END);
+        visit(WhileContext.class).map(WHILE_BEGIN, WHILE_END);
+        visit(RepeatContext.class).map(REPEAT_BEGIN, REPEAT_END);
+
```

Comment:
Do we want to have three separate loop tokens here or combine them into one?
But these changes could also be subject of a different PR.